analyses/ARI-model-test.qmd

```diff
@@ -0,0 +1,279 @@
+---
+title: "Testing the Impact of the Affordability Risk Index on the Model"
+format: html
+---
+```{r, include = FALSE}
+library(sf)
+library(dplyr)
+library(ggplot2)
+library(noctua)
+library(arrow)
+library(kableExtra)
+library(spdep)
+library(leaflet)
+
+
+con <- dbConnect(noctua::athena())
+noctua::noctua_options(cache = 10)
+
+input_data <- read_parquet("input/assessment_data.parquet") %>%
+  select(meta_pin, meta_township_name, meta_nbhd_code, meta_card_num, other_affordability_risk_index, loc_census_acs5_tract_geoid, meta_pin10, loc_longitude, loc_latitude) %>%
+  rename(ARI = other_affordability_risk_index)
+
+working_data <- read_parquet("output/shap/model_shap.parquet") %>%
+  select(meta_pin, other_affordability_risk_index, meta_card_num) %>%
+  rename(ARI_SHAP = other_affordability_risk_index) %>%
+  mutate(pin10 = substr(meta_pin, 1, 10)) %>%
+  inner_join(input_data, by = c("meta_pin", "meta_card_num")) %>%
+  group_by(meta_township_name) %>%
+  mutate(
+    shap_township_mean = mean(abs(ARI_SHAP))
+  ) %>%
+  ungroup() %>%
+  group_by(loc_census_acs5_tract_geoid) %>%
+  mutate(
+    shap_tract_mean = mean(abs(ARI_SHAP))
+  )
+
+working_data_non_zero <- working_data %>%
+  filter(ARI != 0)
+
+# A little hacky, but same as township (used after switching from neighborhood to tract)
+nbhd <- ccao::nbhd_shp
+
+tract_geographies <- dbGetQuery(
+  conn = con,
+  "SELECT *
+   FROM spatial.census
+  where year = '2022' and geography = 'tract'"
+)
+
+cook <- dbGetQuery(
+  conn = con,
+  "SELECT *
+   FROM spatial.census
+  where year = '2022' and geography = 'county'"
+) %>%
+  st_as_sf() %>%
+  filter(name == "Cook")
+
+tract_data <- working_data %>%
+  distinct(loc_census_acs5_tract_geoid, .keep_all = TRUE) %>%
+  inner_join(tract_geographies, by = c("loc_census_acs5_tract_geoid" = "geoid")) %>%
+  st_as_sf()
+
+tract_data_non_zero <- working_data_non_zero %>%
+  distinct(loc_census_acs5_tract_geoid, .keep_all = TRUE) %>%
+  inner_join(tract_geographies, by = c("loc_census_acs5_tract_geoid" = "geoid")) %>%
+  st_as_sf()
+```
+
+
+# Context
+
+The ARI is a tool intended to demonstrate the need to preserve affordability in areas that are becoming less affordable at a faster rate. Census tracts where affordability loss risk is greatest receive the highest scores in this index. The ARI uses American Community Survey data to measure change over time across various factors that indicate affordability loss:
+
+- Median Household Income
+- Median Home Value
+- Families Below the Federal Poverty Level
+- Housing Unit Vacancy
+- Renter Tenancy
+- Individuals Employed in Management, Business, Science and Arts Occupations
+- Individuals with a Four-Year Degree (Bachelor’s) or Higher.
+
+
+![](ARI Image.png)
```

Comment:
**issue:** This image should either be part of the repo (bad) or loaded from its source website (better). Right now, it's not included in the PR so this report can't knit.